corehq/apps/users/views/mobile/groups.py

```diff
@@ -20,6 +32,79 @@ def _get_sorted_groups(domain):
     )
 
 
+def get_group_or_404(domain, group_id):
+    try:
+        group = Group.get(group_id)
+        assert group.doc_type.startswith('Group')
+        assert group.domain == domain
```

Comment:
Using an assert for a check like this could be dangerous, as some Python run modes will strip assert statements during compilation.